executor/joiner.go

```diff
@@ -104,6 +104,9 @@ type joiner interface {
 	// parameter passed to `onMissMatch`.
 	onMissMatch(hasNull bool, outer chunk.Row, chk *chunk.Chunk)
 
+	// onMatch operates on the matched outer row according to the join type.
+	onMatch(outer chunk.Row, chk *chunk.Chunk)
```

Comment:
Why do we need this interface, can we use the `tryToMatchOuters`?

Reply:
@likzn Kindly ping that can we reuse `tryToMatchOuters` instead of create a new method?

Reply:
`tryToMatchOuters ` is already used now.  `onMatch` and `onMissMatch` is a pair. They will used in `probe` to deal with match/missMatch rows.
```

// Concurrently handling rows from the hash table
func (e *HashJoinExec) handleRowsFromHashTable(workerID uint) {
	ok, joinResult := e.getNewJoinResult(workerID)
	if !ok {
		return
	}
	numChks := e.rowContainer.NumChunks()
	for i := int(workerID); i < numChks; i += int(e.concurrency) {
		chk, err := e.rowContainer.GetChunk(i)
		if err != nil {
			// Catching the error and send it
			joinResult.err = err
			e.joinResultCh <- joinResult
			return
		}
		for j := 0; j < chk.NumRows(); j++ {
			if e.outerMatchedStatus[i].UnsafeIsSet(j) {
				// To satisfy property of semi join, we unify process matched rows to avoid duplicate row
				e.joiners[workerID].onMatch(chk.GetRow(j), joinResult.chk)
			} else {
				// Process unmatched outer rows
				e.joiners[workerID].onMissMatch(e.outerUnMatchedIsNull[i].UnsafeIsSet(j), chk.GetRow(j), joinResult.chk)
			}
			if joinResult.chk.IsFull() {
				e.joinResultCh <- joinResult
				ok, joinResult = e.getNewJoinResult(workerID)
				if !ok {
					return
				}
			}
		}
	}

	if joinResult == nil {
		return
	} else if joinResult.err != nil || (joinResult.chk != nil && joinResult.chk.NumRows() > 0) {
		e.joinResultCh <- joinResult
	}
}

```